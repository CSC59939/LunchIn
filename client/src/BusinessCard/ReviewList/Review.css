.review-card{
  display: flex;
  width: 100%;
  height: 100%;
}
.review-card-media{
  object-fit: cover;
  width: 100%;
  height: 100%;
}
.review-content{
  padding: 1%;
  overflow: auto;
}
