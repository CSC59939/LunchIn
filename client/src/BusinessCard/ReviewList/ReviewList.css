.review-list{
  overflow: auto;
  width: 100%;
  height: 100%;
}
.review-item{
  width: 100%;
  height: 100%;
  padding: 0%;
}
