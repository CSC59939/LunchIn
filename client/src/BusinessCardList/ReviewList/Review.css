.review-card{

}
.review-card-media{
  width: 200px;
  height: 150px;
}
