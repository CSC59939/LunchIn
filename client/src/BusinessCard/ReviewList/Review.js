import React, { Component } from "react";
import Card from '@material-ui/core/Card';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import "./Review.css";

class Review extends Component {
  render() {
    if(this.props.review){
      const r = this.props.review;
      return (
        <Card className="review-card">
            <Grid container>
              <Grid item xs={4} sm={4} md={4} lg={4} xl={4}>
                <img
                  className="review-card-media"
                  src={r.user.image_url}
                  alt="Failed to load review URL."
                />
              </Grid>
              <Grid item xs={8} sm={8} md={8} lg={8} xl={8}
                className="review-content"
                >
                <Typography variant="h6" component="h6">
                  {r.user.name}
                </Typography>
                <Typography component="p">
                  {r.time_created.split(' ')[0]} Rating: {r.rating} <br />
                  {r.text}
                </Typography>
              </Grid>
            </Grid>
        </Card>
      );
    }
    else{
      return (<CircularProgress />);
    }
  }
}
export default Review;
