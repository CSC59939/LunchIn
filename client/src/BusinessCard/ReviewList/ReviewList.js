import React, { Component } from "react";
// import Card from '@material-ui/core/Card';
// import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
// import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import CircularProgress from '@material-ui/core/CircularProgress';
import Review from './Review';
import "./ReviewList.css";


class ReviewList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reviewList: null,
      loading: false
    };
  }
  componentDidMount(){
    this.setState({ loading: true });
    fetch(
      `/api/yelp/reviews?id=${this.props.selectedBusinessId}`
    )
    .then(response => response.json())
    .then(data => {
      this.setState({ reviewList: data.reviews });
      this.setState({ loading: false });
    })
    .catch(e => {
      this.setState({ loading: false });
      console.log(e);
    });
  }
  renderReview(review) {
    return (
      <ListItem className="review-item">
        <Review review={review} key={review.id} />
      </ListItem>
    );
  }
  render() {
    return (
      <List component="nav" className="review-list">
        { this.state.reviewList
          ? this.state.reviewList.map(this.renderReview)
          : ( this.state.loading
              ? (<CircularProgress />)
              : (null)
          )
        }
      </List>
    );
  }
}
export default ReviewList;
