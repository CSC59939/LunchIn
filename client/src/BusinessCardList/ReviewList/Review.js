import React, { Component } from "react";
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
import "./Review.css";

class Review extends Component {
  render() {
    if(this.props.review){
      return (
        <Card className="review-card">
            <CardMedia
              className="review-card-media"
              image={this.props.review.user.image_url}
              title={this.props.review.user.image_url}
            />
            <Typography component="p">
              {this.props.review.user.name} <br />
              {this.props.review.time_created} <br />
              {this.props.review.rating} <br />
              {this.props.review.text} <br />
            </Typography>
        </Card>
      );
    }
    else{
      return (<CircularProgress />);
    }
  }
}
export default Review;
