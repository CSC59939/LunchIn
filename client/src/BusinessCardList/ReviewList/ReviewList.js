import React, { Component } from "react";
// import Card from '@material-ui/core/Card';
// import CardActionArea from '@material-ui/core/CardActionArea';
// import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
// import Typography from '@material-ui/core/Typography';
import Review from './Review';
import "./ReviewList.css";


class ReviewList extends Component {
  constructor(props) {
    super(props);
    this.state = { reviewList: [] };
  }
  componentDidMount(){
    fetch(
      `/api/yelp/reviews?id=${this.props.selectedBusinessId}`
    )
    .then(response => response.json())
    .then(data => {
      console.log(data);
      this.setState({ reviewList: data.reviews });
    })
    .catch(e => console.log(e));
  }
  renderReview(review) {
    return (
      <Review review={review} key={review.id} />
    );
  }
  render() {
    return (
      <div className="review-card-list">
        {this.state.reviewList.map(this.renderReview)}
      </div>
    );
  }
}
export default ReviewList;
