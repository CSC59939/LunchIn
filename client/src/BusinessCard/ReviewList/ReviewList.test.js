import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import ReviewList from "./ReviewList";
import toJson, { shallowToJson, mountToJson, renderToJson } from 'enzyme-to-json';

describe("ReviewList Component", () => {
  describe("When provided with no info", () => {
    it("Should render correctly", () => {
      const businessListInstance = shallow(
        <ReviewList />
      );
      expect(shallowToJson(businessListInstance)).toMatchSnapshot();
    });
  });
  describe("When provided with selectedBusinessId", () => {
    it("Should render correctly", () => {
      const selectedBusinessId = "E8RJkjfdcwgtyoPMjQ_Olg";
      const businessListInstance = shallow(
        <ReviewList selectedBusinessId={selectedBusinessId} />
      );
      expect(shallowToJson(businessListInstance)).toMatchSnapshot();
    });
  });
});
