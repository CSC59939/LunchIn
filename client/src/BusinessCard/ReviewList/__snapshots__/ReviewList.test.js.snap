// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList Component When provided with no info Should render correctly 1`] = `
<WithStyles(List)
  className="review-list"
  component="nav"
>
  <WithStyles(CircularProgress) />
</WithStyles(List)>
`;

exports[`ReviewList Component When provided with selectedBusinessId Should render correctly 1`] = `
<WithStyles(List)
  className="review-list"
  component="nav"
>
  <WithStyles(CircularProgress) />
</WithStyles(List)>
`;
